### ISSUE TITLE
Inline comments are not formatted properly for Mariadb

### ISSUE THREAD

**Issue by missquote** [sqlfluff/sqlfluff#7486]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

While Mariadb's spec requires a space after the two dashes for inline comments, Sqlfluff's maradb dialect does not appear to require this. No error is flagged and it is not fixed when run with `sqlfluff fix`.

https://mariadb.com/docs/server/reference/sql-statements/comment-syntax

### Expected Behaviour

When encountering a Mariadb inline comment with no space after dashes, sqlfluff should recognize this as an error.

### Observed Behaviour

The malformed comment is not flagged as an error and is not fixed.

### How to reproduce

```
--create whois table
CREATE TABLE whois (
    domain VARCHAR(256) PRIMARY KEY,
    created_at TIMESTAMP(6) NULL DEFAULT NULL,
    expiration TIMESTAMP(6) NULL DEFAULT CURRENT_TIMESTAMP
);
```

### Dialect

Mariadb

### Version

sqlfluff, version 4.0.4
Python 3.11.14

### Configuration

```
[sqlfluff]
dialect = mariadb
exclude_rules = CV08, LT14, RF01, RF02, RF04, RF05
fix_even_unparsable = True
large_file_skip_byte_limit = 0
max_line_length = 120

[sqlfluff:indentation]
implicit_indents = allow

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = upper
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by DoriAlagem** [comment]
I've submitted a fix in PR #7713. It adds a MariaDB-specific `inline_comment` lexer override that requires whitespace after `--` (using a `(?=\s)` lookahead), matching the MariaDB spec. MySQL behavior is unchanged. Tests confirm `--comment` (no space) produces a `PRS` violation in MariaDB but not in MySQL.

**Comment by peterbud** [comment]
* MySql syntax in our parser is permissive, allows '--' and '-- ' for comments in sqlfluff - see #5724 and #5731
* MariaDB should have the same permissive  behaviour, as it states " The space after the two dashes is required (as in MySQL).", and as I understood both system accepts comments with and without spaces after '--'.
